tequilapi/endpoints/service.go

```diff
@@ -261,9 +246,53 @@ func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManage
 	router.DELETE("/services/:id", serviceEndpoint.ServiceStop)
 }
 
-func toServiceRequest(req *http.Request) (sr serviceRequest, err error) {
-	err = json.NewDecoder(req.Body).Decode(&sr)
-	return sr, err
+func (se *ServiceEndpoint) toServiceRequest(req *http.Request) (serviceRequest, error) {
+	var jsonData struct {
+		ProviderID  string           `json:"providerId"`
+		ServiceType string           `json:"serviceType"`
+		Options     *json.RawMessage `json:"options"`
+	}
+	if err := json.NewDecoder(req.Body).Decode(&jsonData); err != nil {
+		return serviceRequest{}, err
+	}
+
+	sr := serviceRequest{
+		ProviderID:  jsonData.ProviderID,
+		ServiceType: se.toServiceType(jsonData.ServiceType),
+		Options:     se.toServiceOptions(jsonData.ServiceType, jsonData.Options),
+	}
+	return sr, nil
+}
+
+func (se *ServiceEndpoint) toServiceType(value string) string {
+	if value == "" {
+		return ""
+	}
+
+	_, ok := se.optionsParser[value]
+	if !ok {
+		return serviceTypeInvalid
+	}
+
+	return value
+}
+
+func (se *ServiceEndpoint) toServiceOptions(serviceType string, value *json.RawMessage) service.Options {
+	if value == nil {
+		return nil
+	}
+
+	optionsParser, ok := se.optionsParser[serviceType]
+	if !ok {
+		return nil
+	}
+
+	options, err := optionsParser(*value)
+	if err != nil {
+		return serviceOptionsInvalid
```

Comment:
I'd prefer to return `err` here, to give more details of what was invalid in the request.

Reply:
I did not degrade - there was no error also. Created separate ticket to improve that and more #763